eng/Directory.Build.Common.props

```diff
@@ -116,14 +116,14 @@
     <SupportsNetStandard20 Condition="'$(SupportsNetStandard20)' == '' and '$(IsClientLibrary)' == 'true'">true</SupportsNetStandard20>
     <SupportsNetStandard20 Condition="'$(SupportsNetStandard20)' == ''">false</SupportsNetStandard20>
 
-    <RequiredTargetFrameworks>net452;netstandard1.4;net461;netstandard2.0</RequiredTargetFrameworks>
-    <RequiredTargetFrameworks Condition="'$(SupportsNetStandard20)' == 'true'">netstandard2.0</RequiredTargetFrameworks>
+    <RequiredTargetFrameworks>net7.0;net452;netstandard1.4;net461;netstandard2.0</RequiredTargetFrameworks>
+    <RequiredTargetFrameworks Condition="'$(SupportsNetStandard20)' == 'true'">net7.0;netstandard2.0</RequiredTargetFrameworks>
```

Comment:
This impacts all libraries; what's your test plan for this change?

Reply:
I'm working on this and all the necessary changes in another PR. There are other issues to resolve, so please do not upgrade this to net7.0 in this PR.

Reply:
Sure. I saw your PR and I am aware that more work needs to be done for bumping to .NET 7. This is for prototyping now. Using static abstract interface method available from .NET 7 is one of the solutions we want to explore.

Reply:
Last I talked to @KrzysztofCwalina and @tg-msft - our architects - late last year, all APIs have to be present in all versions we support with rare exception. Given static interface members require runtime changes, please talk with them.

Even if accepted, this must still wait. Updating all projects just so some can use net7.0 will have adverse effects, especially given that 7.0.100 regresses important behavior that will greatly increase build times.

Reply:
Yes, when I talked to Michael about it initially, we discussed using this only as an implementation detail, i.e. we cannot have public API differences as it's viral and prevents customers from building portable libraries.

Reply:
Why multi-target just to use an implementation detail that we'd have to accommodate targeting netstandard2.0 anyway? Is the "work around" for netstandard2.0 less performant such that we need to multi-target? This adds complexity and, though minor, does increase nupkg size: build times in e.g., CIs, will increase as a result.